packages_rs/nextclade/src/analyze/find_aa_motifs_changes.rs

```diff
@@ -0,0 +1,153 @@
+pub type AaMotifsChangesMap = BTreeMap<String, AaMotifChanges>;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AaMotifChanges {
+  pub preserved: Vec<AaMotifMutation>,
+  pub gained: Vec<AaMotifMutation>,
+  pub lost: Vec<AaMotifMutation>,
+  pub total: usize,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Ord, PartialOrd, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct AaMotifMutation {
+  pub name: String,
+  pub gene: String,
+  pub position: usize,
+  pub ref_seq: String,
+  pub qry_seq: String,
+}
+
+/// Find changes between AA motifs in reference and in query sequence
+pub fn find_aa_motifs_changes(
+  aa_motifs_ref: &AaMotifsMap,
+  aa_motifs_qry: &AaMotifsMap,
+  ref_peptides: &TranslationMap,
+  translations: &[Translation],
+) -> Result<AaMotifsChangesMap, Report> {
+  zip_map_hashmap(aa_motifs_ref, aa_motifs_qry, |name, motifs_ref, motifs_qry| {
+    let changes = find_aa_motifs_changes_one(motifs_ref, motifs_qry, ref_peptides, translations)?;
+    Ok((name.clone(), changes))
+  })
+  .collect()
+}
+
+fn find_aa_motifs_changes_one(
+  motifs_ref: &[AaMotif],
+  motifs_qry: &[AaMotif],
+  ref_peptides: &TranslationMap,
+  translations: &[Translation],
+) -> Result<AaMotifChanges, Report> {
+  // We want to find added and removed positions (`.pos`) per gene.
+  // So we use a wrapper for `struct AaMotif` which disregards `.seq` during comparison.
+  let motifs_ref: HashSet<AaMotifWithoutSeq> = motifs_ref.iter().cloned().map(AaMotifWithoutSeq::from).collect();
+  let motifs_qry: HashSet<AaMotifWithoutSeq> = motifs_qry.iter().cloned().map(AaMotifWithoutSeq::from).collect();
+
+  // Gained motifs: not present in ref, present in qry
+  let gained = motifs_qry
+    .difference(&motifs_ref)
+    .map(|motif| add_ref_seq(&motif.0, ref_peptides))
+    .collect::<Result<Vec<AaMotifMutation>, Report>>()?
+    .into_iter()
+    .sorted()
+    .collect_vec();
+
+  // Lost motifs: present in ref, not present in query
+  let lost = motifs_ref
+    .difference(&motifs_qry)
+    .map(|motif| add_qry_seq(&motif.0, translations))
+    .collect::<Result<Vec<AaMotifMutation>, Report>>()?
+    .into_iter()
+    .sorted()
+    .collect_vec();
+
+  // Preserved motifs: present in ref and qry
+  let preserved = {
+    // Zip ref and qry motifs, so that we can compare them pairwise
+    let kept_ref: Vec<AaMotif> = cloned_into(motifs_ref.intersection(&motifs_qry)).sorted().collect_vec();
+    let kept_qry: Vec<AaMotif> = cloned_into(motifs_qry.intersection(&motifs_ref)).sorted().collect_vec();
+    Zip::from((kept_ref, kept_qry))
+      .map(|(motif_ref, motif_qry)| AaMotifMutation {
+        name: motif_ref.name,
+        gene: motif_ref.gene,
+        position: motif_ref.position,
+        ref_seq: motif_ref.seq,
+        qry_seq: motif_qry.seq,
+      })
+      .sorted()
+      .collect_vec()
+  };
+
+  let total = gained.len() + lost.len() + preserved.len();
```

Comment:
I think the `total` should be the just those in the query. that is `gained + preserved`.

Reply:
Fixed in f4c598b2b19c46411f7c3afc5b7cde363644e468